autoload/ale/handlers.vim

```diff
@@ -220,3 +220,49 @@ function! ale#handlers#HandleStyleLintFormat(buffer, lines) abort
 
     return l:output
 endfunction
+
+function! ale#handlers#HandleGhcFormat(buffer, lines) abort
+    " Look for lines like the following.
+    "
+    " /dev/stdin:28:26: Not in scope: `>>>>>'
+     "Appoint/Lib.hs:8:1: warning:
+    let l:pattern = '^[^:]\+:\(\d\+\):\(\d\+\):\s\{-}\(warning\|error\)\(.\+\)$'
```

Comment:
Can we be confident about the changes to the pattern here? I don't use GHC myself. As long as we're sure it will work with the output of GHC across different versions and so on, then I'm happy.

There are a few test cases below, and that's good. :+1: